homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -202,6 +228,15 @@ async def async_step_user(
         if user_input.get(CONF_VOLUMES):
             config_data[CONF_VOLUMES] = user_input[CONF_VOLUMES]
 
+        if existing_entry and self.reauth_conf:
+            self.hass.config_entries.async_update_entry(
+                existing_entry, data=config_data
+            )
+            await self.hass.config_entries.async_reload(existing_entry.entry_id)
```

Comment:
Why not always do this regardless if reauth or not?

Reply:
If you come here via discovery, it might have an updated host.

Reply:
for this PR I wanted to limit the scope to the re-authentication (_in case user changed credentials_)

on the other hand, I already thinking about a re-configuration flow (_change hostname/ip, port, ssl, ..._), but still struggle with, how to consider the NAS is not reachable, since settings were changed or it is only rebooting :thinking: 
For now I see these options:
- user can simple add the same NAS again (_identified via its serial_), the configflow will recognize this and update the already existing config-entry
- the settings host/ip, port and ssl will be available to be changed via the options-flow

@balloob what do you think? Do I miss here any already existing approach to handle re-configurations of - let's say - basic settings?

Reply:
If you can fetch the serial without authentication, you could automatically update the data when it's discovered without requiring any work from the user. Fine with leaving it to a future PR.

Reply:
many thanks for the hints. I have created a new PR #53285 for that